### ISSUE TITLE
Automatically count revisions based on Git commit count

### ISSUE THREAD

**Issue by chrisfenner** [TrustedComputingGroup/markdown#45]
(Suggestion by Andreas)

Idea would be (in bash)

```bash
gitver=($(git describe --tags --always | tr '-' ' '))

specversion=”${gitver[0]}”

if [ “$(git branch)” = “main” ] then

specrevision=”${gitver[1]}”

else

specrevision=”${gitver[1]}-${gitver[2]}”

fi
```

Such that on the main branch we would only ever have version and revision (as number of commits since tag), but on branches where revision is not unique the commit id gets added as well.

**Comment by AndreasFuchsTPM** [comment]
Just a P.S.
Of course the downside is that you can only have a single spec per repo, because the tag affects the whole repo.
An approach with multiple specs in a repo (e.g. in different directories) is not possible with this.

**Comment by chrisfenner** [comment]
Yeah. I think for now, we should pursue just one spec per repo anyway.

**Comment by williamcroberts** [comment]
@chrisfenner looks like the plumbing to support this is mostly there. But I can't say I Grok all the details. I see that Pandoc supports `--metadata` is that allow for variables to be resolved in a YAML metadata block (The opening block in the markdown delimited with `---` and `...`) or is that YAML block used for resolving variables elsewhere? If we could just add version to the metadata using git describe output that would be easy to put together.

I see references to things like `/resources/img/cover.png` and `eisvogel.latex` where do those originate from?

**Comment by chrisfenner** [comment]
That's right, I think that YAML block can be used for any variable referenced from the latex in the way we are already using it.

Those resources are in github.com/trustedcomputinggroup/pandoc and bundled into the docker container (since setting up the dependencies every time really stinks).

**Comment by williamcroberts** [comment]
> That's right, I think that YAML block can be used for any variable referenced from the latex in the way we are already using it.
> 
The references expand from within the LaTeX template, is that right? So in the case we want to set them via `--metadata` we would omit them from the YAML block within the markdown?

> Those resources are in github.com/trustedcomputinggroup/pandoc and bundled into the docker container (since setting up the dependencies every time really stinks).

Great, that's what I figured just wasn't sure where they were :+1:

**Comment by chrisfenner** [comment]
Oh, right. Anything we want to send to the tool from the GH action layer we can use `--metadata` for. My intention is to let the YAML block be obvious things that the workgroups edit. So, for this feature I think we would do the git-fu from the GH action and pass the values with --metadata.

**Comment by williamcroberts** [comment]
> Oh, right. Anything we want to send to the tool from the GH action layer we can use `--metadata` for. My intention is to let the YAML block be obvious things that the workgroups edit. So, for this feature I think we would do the git-fu from the GH action and pass the values with --metadata.

Yep my thoughts exactly, and just to get the last part of my question, those variables are resolved in the latex template in the pandoc repo like version here https://github.com/TrustedComputingGroup/pandoc/blob/main/template/eisvogel.latex#L897?

I think I'll let folks pass a variable to the workflow to opt-into git versioning of documents, thoughts?

**Comment by chrisfenner** [comment]
Yep, that's how the variables get consumed. BTW I am not a Latex expert, I have just gotten something working. If you see something that needs fixed on that level, please let me know!

> I think I'll

If that implies you will send a PR to add this, you can do this however you like. 😅 I agree with your approach.

**Comment by williamcroberts** [comment]
> Yep, that's how the variables get consumed. BTW I am not a Latex expert, I have just gotten something working. If you see something that needs fixed on that level, please let me know!
> 
> > I think I'll
> 
> If that implies you will send a PR to add this, you can do this however you like. sweat_smile I agree with your approach.

Ha yeah, Ill probably try and do something next week as I start back on spec work.
